### PR TITLE
fix(release): support customized default branch

### PR DESCRIPTION
**Pull request by yj-yan**
Please read [CONTRIBUTING.md](CONTRIBUTING.md) for additional information on contributing to this repository!

## What this PR does / why we need it
Do not hardcode the default branch since some repos uses `master` as default. Also added quotes to release branch for consistency.

## Jira ID

[XX-XX]

### CODE REVIEW

File: templates/main_test.go [hunk 1]
```diff
@@ -30,6 +30,17 @@ func TestConfigForAutoPrerelease(t *testing.T) {
 	st.Run(stenciltest.RegenerateSnapshots())
 }
 
+func TestConfigForDisabledPrerelease(t *testing.T) {
+	st := stenciltest.New(t, ".circleci/config.yml.tpl", "_helpers.tpl")
+	st.Args(map[string]interface{}{
+		"releaseOptions": map[string]interface{}{
+			"enablePrereleases": false,
+			"prereleasesBranch": "",
```
[the remark this answers is not in the document]
Comment (yj-yan) [reply templates/main_test.go:38]: This seems to be the default settings for repos without prereleases enabled, so I just want to add it into the test case.

[the remark this answers is not in the document]
Comment (malept) [reply templates/main_test.go:38]: So I would do one of two things here:

1. Omit it from the tests
2. Add a comment to the test explaining that this is normally an implicit value

The rationale is that a lot of the time, if an option isn't documented well enough, people will generally go to the tests to see how it's expected to be used. I don't particularly want to imply to folks that they need to specify an empty `prereleasesBranch` value if they don't want prereleases enabled.

File: templates/.circleci/config.yml.tpl [hunk 2]
```diff
@@ -63,12 +63,12 @@ test: &test
...
 # Branches used for releasing code, pre-release or not
 release_branches: &release_branches
-  {{- $stableBranch := "main" }}
+  {{- $stableBranch := $defaultBranch }}
   {{- if $prereleases }}
   {{- $pb := stencil.Arg "releaseOptions.prereleasesBranch" }}
   {{- $stableBranch = "release" }}
   # Release branch
-  - release
+  - {{ $stableBranch | squote }}
```
[the remark this answers is not in the document]
Comment (malept) [reply templates/.circleci/config.yml.tpl:71]: I would make this use double quotes and then make a followup PR to change the rest of the `squote` references.

File: templates/.circleci/config.yml.tpl [hunk 3]
```diff
@@ -124,7 +124,7 @@ workflows:
     when: << pipeline.parameters.rebuild_cache >>
     jobs:
       - shared/save_cache: *test
-  {{- if stencil.Arg "releaseOptions.autoPrereleases" }}
+  {{- if and (stencil.Arg "releaseOptions.enablePrereleases") (stencil.Arg "releaseOptions.autoPrereleases") }}
```
Comment (yj-yan) [inline templates/.circleci/config.yml.tpl:127]: Added this check to ensure if `enablePrereleases` is disabled, the autoPrereleases will be ignored.

Comment (malept) [reply templates/.circleci/config.yml.tpl:127]: Do we need extra tests for this now that there's two options needed?

Comment (yj-yan) [reply templates/.circleci/config.yml.tpl:127]: I briefly tested it but the two options are mutual. I will add the test case, and comment it as only to verify the corner case.
